pkg/operator/staticresource/controller.go

```
@@ -0,0 +1,289 @@
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isManaged(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == "pod-identity-webhook" && meta.GetNamespace() == operatorNamespace {
+		return true
+	}
+	// we also care about service-ca rotation
+	if meta.GetName() == "service-ca" && meta.GetNamespace() == "openshift-config-managed" {
+		return true
+	}
+	return false
+}
+
+type staticResourceReconciler struct {
+	clientset                              *kubernetes.Clientset
+	logger                                 log.FieldLogger
+	eventRecorder                          events.Recorder
+	deploymentGeneration                   int64
+	mutatingWebhookConfigurationGeneration int64
+	forceRollout                           bool
+}
+
+var _ reconcile.Reconciler = &staticResourceReconciler{}
+
+func (r *staticResourceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.WithField("controller", controllerName).Debugf("reconciling after watch event %#v", request)
+	if request.Name == "service-ca" && request.Namespace == "openshift-config-managed" {
+		log.WithField("controller", controllerName).Infof("%s configmap change detected", request.NamespacedName)
+		r.forceRollout = true
+	}
+	r.ReconcileResources()
+	return reconcile.Result{}, nil
+}
+
+func (r *staticResourceReconciler) ReconcileResources() {
+	applyResults := resourceapply.ApplyDirectly(
+		(&resourceapply.ClientHolder{}).WithKubernetes(r.clientset),
+		r.eventRecorder,
+		v410_00_assets.Asset,
+		staticFiles...,
+	)
+	for _, result := range applyResults {
+		if result.Error != nil {
+			log.WithField("controller", controllerName).WithError(result.Error).Errorf("error reconciling %s", result.Type)
+			continue
```

Should we error rather than continue if we've failed to apply some required assets?